libs/base/Data/Vect.idr

```diff
@@ -18,15 +20,14 @@ data Vect : (len : Nat) -> (elem : Type) -> Type where
 %name Vect xs,ys,zs,ws
 
 public export
-length : (xs : Vect len elem) -> Nat
-length [] = 0
-length (x::xs) = 1 + length xs
+length : {1 len : Nat} -> (0 xs : Vect len elem) -> Nat
```

Comment:
This looks like a breaking change since `len` is made non-erasable. Perhaps we should have two separate `length` functions.

Reply:
Yes, `length` shouldn't be written like this because it will leak the length everywhere. I don't see a need for a second `length` function where you already have the length available. Perhaps if someone comes up with one, we'll add it.

Reply:
What's the use case for a `length` function for `Vect`? A `Vect` is already wearing a name tag that says "Hello, my length is ___", so why would its length need to be recursively calculated?

Reply:
I changed it back, in any case.

Reply:
> so why would its length need to be recursively calculated?

The length is known at *compile* time but it is not stored at *run* time.
If you need to access the length at run time then you need to compute it
(or pass around a pair `(n ** Vec n a)`.